### PR TITLE
Issue#96235 Add POA Request Migrations

### PR DESCRIPTION
**Pull request by ojbucao**
This PR lays the groundwork for managing Power of Attorney request lifecycle events such as resolutions, replacements, withdrawals, expirations, and decisions, ensuring seamless integration with the vets-api monolith while maintaining namespace isolation for the engine.

This PR introduces the foundational database and framework changes for managing Power of Attorney requests within the accredited_representative_portal module. Key updates include:

Migrations
• Added migrations to create the following tables with UUIDs:
• ar_power_of_attorney_request_resolutions: Includes resolving_type, resolving_id, reason_ciphertext, and timestamps.
• ar_power_of_attorney_request_replacements
• ar_power_of_attorney_request_withdrawals
• ar_power_of_attorney_request_expirations
• ar_power_of_attorney_request_decisions: Includes a type field and a foreign key reference to user_accounts for created_by.
• ar_power_of_attorney_requests
• ar_power_of_attorney_forms: Includes blind indexes (city_bidx, state_bidx, zipcode_bidx) with concurrent indexing.

Depends on: https://github.com/department-of-veterans-affairs/vets-api/pull/19881

### CODE REVIEW

File: modules/accredited_representative_portal/db/migrate/20241212000000_create_ar_power_of_attorney_request_resolutions.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+    create_table :ar_power_of_attorney_request_replacements, id: :uuid
+
+    create_table :ar_power_of_attorney_request_withdrawals, id: :uuid
+
+    create_table :ar_power_of_attorney_request_expirations, id: :uuid
+
+    create_table :ar_power_of_attorney_request_decisions, id: :uuid do |t|
+      t.string 'type', null: false
+      t.references 'created_by', type: :uuid, foreign_key: { to_table: :user_accounts }
```
Comment (nihil2501) [inline modules/accredited_representative_portal/db/migrate/20241212000000_create_ar_power_of_attorney_request_resolutions.rb:20]: @ojbucao I eventually thought `creator_id` might be more typical naming,.

File: modules/accredited_representative_portal/db/migrate/20241212000000_create_ar_power_of_attorney_request_resolutions.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    create_table :ar_power_of_attorney_request_resolutions, id: :uuid do |t|
+      t.string 'resolving_type', null: false
+      t.uuid 'resolving_id', null: false
+      t.text 'reason_ciphertext'
+      t.datetime 'created_at', null: false
+    end
+
+    create_table :ar_power_of_attorney_request_replacements, id: :uuid
+
+    create_table :ar_power_of_attorney_request_withdrawals, id: :uuid
```
Comment (nihil2501) [inline modules/accredited_representative_portal/db/migrate/20241212000000_create_ar_power_of_attorney_request_resolutions.rb:14]: I think we can omit replacements and withdrawals for the time being.

We don't really know what product thinks about submitting a new POA request when another one already exists in our system. Until we broach this question with product, we can have no logic here and just allow a new POA request creation.

Similarly for withdrawals, I have not heard any discussion of allowing for the claimant to withdraw a given POA request.

File: modules/accredited_representative_portal/db/migrate/20241212000100_create_ar_power_of_attorney_requests.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+class CreateArPowerOfAttorneyRequests < ActiveRecord::Migration[7.1]
+  def change
+    create_table :ar_power_of_attorney_requests, id: :uuid do |t|
+      t.references :claimant, type: :uuid, foreign_key: { to_table: :user_accounts }
+      t.references :latest_resolution, type: :uuid, foreign_key: { to_table: :ar_power_of_attorney_request_resolutions }
```
Comment (nihil2501) [inline modules/accredited_representative_portal/db/migrate/20241212000100_create_ar_power_of_attorney_requests.rb:5]: Let's make POA request have one resolution in the app and leave out this denormalization for now. This was that discussion where I said a future migration won't be too painful probably.

File: modules/accredited_representative_portal/db/migrate/20241212000300_add_indexes_to_ar_power_of_attorney_forms.rb [hunk 4]
```diff
@@ -0,0 +1,8 @@
+class AddIndexesToArPowerOfAttorneyForms < ActiveRecord::Migration[7.1]
+  disable_ddl_transaction!
+  def change
+    add_index :ar_power_of_attorney_forms, :city_bidx, algorithm: :concurrently
+    add_index :ar_power_of_attorney_forms, :state_bidx, algorithm: :concurrently
+    add_index :ar_power_of_attorney_forms, :zipcode_bidx, algorithm: :concurrently
```
Comment (nihil2501) [inline modules/accredited_representative_portal/db/migrate/20241212000300_add_indexes_to_ar_power_of_attorney_forms.rb:6]: I think we probably don't need it, but can you spend a little time convincing yourself whether or not we especially need [multicolumn indexes](https://www.postgresql.org/docs/current/indexes-multicolumn.html) to support our queries? Even if we get this wrong, it's no big deal, but good just to have it somewhere in the back of your mind.

File: modules/accredited_representative_portal/db/migrate/20241212000200_create_ar_power_of_attorney_forms.rb [hunk 5]
```diff
@@ -0,0 +1,14 @@
+class CreateArPowerOfAttorneyForms < ActiveRecord::Migration[7.0]
+  def change
+    create_table :ar_power_of_attorney_forms, id: :uuid do |t|
+      t.references :ar_power_of_attorney_request,
+        type: :uuid,
+        foreign_key: true,
+        index: { name: 'index_ar_poa_forms_on_ar_poa_request_id' }
```
Comment (nihil2501) [inline modules/accredited_representative_portal/db/migrate/20241212000200_create_ar_power_of_attorney_forms.rb:7]: `add_reference` indexes by [default](https://api.rubyonrails.org/classes/ActiveRecord/ConnectionAdapters/SchemaStatements.html#method-i-add_reference). Do we need this? Was it index name length that prompted the manual specification here?

File: modules/accredited_representative_portal/db/migrate/20241212000000_create_ar_power_of_attorney_request_resolutions.rb [hunk 6]
```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+class CreateArPowerOfAttorneyRequestResolutions < ActiveRecord::Migration[7.1]
+  def change
+    create_table :ar_power_of_attorney_request_resolutions, id: :uuid do |t|
```
Comment (nihil2501) [inline modules/accredited_representative_portal/db/migrate/20241212000000_create_ar_power_of_attorney_request_resolutions.rb:5]: We need to have this reference POA requests:
> resolution belongs_to poa_request
> poa_request has_one resolution

**Review by nihil2501 — CHANGES_REQUESTED** [review]

File: modules/accredited_representative_portal/db/migrate/20241212000200_create_ar_power_of_attorney_forms.rb (hunk 5 above)
Comment (ojbucao) [reply modules/accredited_representative_portal/db/migrate/20241212000200_create_ar_power_of_attorney_forms.rb:7]: yeah the length went over the limit

File: modules/accredited_representative_portal/db/migrate/20241212000300_add_indexes_to_ar_power_of_attorney_forms.rb (hunk 4 above)
Comment (ojbucao) [reply modules/accredited_representative_portal/db/migrate/20241212000300_add_indexes_to_ar_power_of_attorney_forms.rb]: ok i changed it to multi-column and a single one for zip

File: modules/accredited_representative_portal/db/migrate/20241212000200_create_ar_power_of_attorney_forms.rb [hunk 7]
```diff
@@ -0,0 +1,14 @@
+class CreateArPowerOfAttorneyForms < ActiveRecord::Migration[7.0]
+  def change
+    create_table :ar_power_of_attorney_forms, id: :uuid do |t|
+      t.references :ar_power_of_attorney_request,
```
Comment (nihil2501) [inline modules/accredited_representative_portal/db/migrate/20241212000200_create_ar_power_of_attorney_forms.rb:4]: `power_of_attorney_request`

Comment (nihil2501) [inline modules/accredited_representative_portal/db/migrate/20241212000200_create_ar_power_of_attorney_forms.rb:4]: NOT NULL

File: modules/accredited_representative_portal/db/migrate/20241212000100_create_ar_power_of_attorney_requests.rb [hunk 8]
```diff
@@ -0,0 +1,8 @@
+class CreateArPowerOfAttorneyRequests < ActiveRecord::Migration[7.1]
+  def change
+    create_table :ar_power_of_attorney_requests, id: :uuid do |t|
+      t.references :claimant, type: :uuid, foreign_key: { to_table: :user_accounts }
+      t.datetime :created_at
```
Comment (nihil2501) [inline modules/accredited_representative_portal/db/migrate/20241212000100_create_ar_power_of_attorney_requests.rb:5]: NOT NULL

File: modules/accredited_representative_portal/db/migrate/20241212000150_create_ar_power_of_attorney_request_resolutions.rb [hunk 9]
```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+class CreateArPowerOfAttorneyRequestResolutions < ActiveRecord::Migration[7.1]
+  def change
+    create_table :ar_power_of_attorney_request_resolutions, id: :uuid do |t|
+      t.string 'resolving_type', null: false
+      t.uuid 'resolving_id', null: false
+      t.text 'reason_ciphertext'
+      t.datetime 'created_at', null: false
+      t.references 'poa_request', type: :uuid, foreign_key: { to_table: :ar_power_of_attorney_requests }
```
Comment (nihil2501) [inline modules/accredited_representative_portal/db/migrate/20241212000150_create_ar_power_of_attorney_request_resolutions.rb:10]: NOT NULL

**Review by nihil2501 — CHANGES_REQUESTED** [review]
Need to fix:
- NOT NULLs
- typos/consistent naming
- need to enforce `has_one` by adding uniqueness to indexes
  - poa request has one form
  - poa request has one resolution
  - for any instance of a resolution sub type there is only one resolution pointing to it
    - i think that would be `add_index :ar_power_of_attorney_request_resolutions, [:resolving_type, :resolving_id], unique: true`

File: modules/accredited_representative_portal/db/migrate/20241212000200_create_ar_power_of_attorney_forms.rb [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+    create_table :ar_power_of_attorney_forms, id: :uuid do |t|
+      t.references :power_of_attorney_request,
+        type: :uuid,
+        foreign_key: { to_table: :ar_power_of_attorney_requests },
+        index: { name: 'index_ar_poa_forms_on_ar_poa_request_id' },
+        null: false
+      t.text :data_ciphertext
+      t.string :city_bidx
+      t.string :state_bidx
+      t.string :zipcode_bidx
```
Comment (nihil2501) [inline modules/accredited_representative_portal/db/migrate/20241212000200_create_ar_power_of_attorney_forms.rb:12]: oh maybe these are null false too

**Review by nihil2501 — APPROVED** [review]
LGTM!

final change i noticed, maybe all of forms columns should be NOT NULL?

**Review by nihil2501 — APPROVED** [review]
LGTM

**Review by nihil2501 — APPROVED** [review]
LGTM

**Review by nihil2501 — APPROVED** [review]
LGTM

**Pull request merged**